grammar.js

```diff
@@ -159,7 +159,7 @@ module.exports = grammar({
 
         init: $ => choice(
             seq('{', '}'),
-            seq('{', repeat(seq($.expr, ',')), $.expr, '}'),
+            seq('{', repeat(seq($.expr, ',')), optional($.expr), '}'),
```

Comment:
Would `seq('{', optional(list1($.expr, ',', true), '}')` work here as well? That feels slightly more expressive than two matchers which could both match nothing to me (though it that doesn't seem to introduce ambiguities here).

Reply:
Yep, works! I've force-pushed the change.

I noticed that re-generating the parser only changed grammar.js, while parser.c was left alone. Interesting. Normally changes affect both.

Reply:
> I noticed that re-generating the parser only changed grammar.js, while parser.c was left alone. Interesting. Normally changes affect both.

The generated `grammar.js` is probably directly derived from the source `grammar.js`, so that it changes seems natural. I suspect `parser.c` does not change since a sequence of `optional` gets transformed into a set of `choice` but I haven't checked (would seem natural from a parser state machine POV).